controller/src/constants.rs

```diff
@@ -15,3 +17,20 @@ pub(crate) fn requeue_slow() -> Action {
 pub(crate) fn no_requeue() -> Action {
     Action::await_change()
 }
+
+/// Parse a duration string into a Duration object.
+pub(crate) fn parse_duration(input: &str) -> Result<Duration, ParseIntError> {
+    let mut secs: u64 = 0;
+    let mut duration_string = input;
+    let units = vec![('d', 86400), ('h', 3600), ('m', 60)];
```

Comment:
I added the function to the `constants.rs` file because it seemed like that file contained functions used by all controllers. However, I can definitely move it to its own file if that would be more idiomatic.

Reply:
It's fine to leave it in constants.rs, probably better than adding a new file or renaming a file for one function that doesn't "fit"